### PR TITLE
mozjpeg: add version 4.0.0

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **mozjpeg/4.0.0**

They got away with their autotools build system and gone full with cmake.
No patches required anymore (I think)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mozjpeg/all/conanfile.py [hunk 1]
```diff
@@ -129,22 +136,27 @@ def build(self):
...
             cmake.install()
             tools.rmdir(os.path.join(self.package_folder, "doc"))
         else:
             autotools = self._configure_autotools()
             autotools.install()
-            tools.rmdir(os.path.join(self.package_folder, "share"))
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            for la_file in glob.glob(os.path.join(self.package_folder, "lib", "*.la")):
-                os.remove(la_file)
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```
Comment (prince-chrismc) [inline recipes/mozjpeg/all/conanfile.py:146]: Please require min conan version for this new feature

Comment (madebr) [reply recipes/mozjpeg/all/conanfile.py:146]: Done

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
Nice simplification

**Review by SSE4 — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Comment by theirix** [comment]
closes #3563

**Pull request merged**